### DISCUSSION TITLE
How to integrate with Lazy.nvim?

### DISCUSSION THREAD

**Discussion by poperigby** [BirdeeHub/nixCats-nvim#26] [Q&A] [answered]
I use Lazy.nvim for managing all my plugins, and it's very convenient how I can set up lazy load triggers and configuration for each plugin in it's own file. How exactly can I use nixCats-nvim with Lazy.nvim? I see https://github.com/BirdeeHub/nixCats-nvim/blob/main/nix/templates/luaUtils/lua/nixCatsUtils/lazyCat.lua but I don't know how to use it.

**Comment by BirdeeHub** [comment]
[kickstart.nvim implemented using the lazy wrapper with tips along the way as a tutorial](https://github.com/BirdeeHub/nixCats-nvim/blob/main/nix/templates/kickstart-nvim/init.lua)

[:h nixCats.luaUtils.lazy](https://github.com/BirdeeHub/nixCats-nvim/blob/fc0aa779842997fb355ea4b99bc208b3d226ad4d/nix/nixCatsHelp/luaUtils.txt#L59-L112)

**Reply by BirdeeHub** [reply, marked as answer]
also, lz.n just came out and its like lazy.nvim that only does loading and no downloading, so it doesnt need a wrapper. You can just put the thing in optionalPlugin and use lz.n to load it instead

https://github.com/nvim-neorocks/lz.n

Keep in mind, the example in their readme is for the home manger syntax, nixCats actually supports that syntax as well, but it was designed to allow you to do it right in your lua files in your normal lua config, so just stick the plugin in optionalPlugins and then go stick the require('lz.n').load statements somewhere in your config

**Reply by BirdeeHub** [reply]
Oh, also, for the name field for those, check :NixCatsPawsible for the list of names (and their paths in the store) of all the plugins included by nix

**Reply by poperigby** [reply]
Thanks :slightly_smiling_face:

**Comment by BirdeeHub** [comment]
Quick update, the lazy wrapper is simpler now, it only needs the path to lazy as the first arg and the regular arguments to lazy.nvim setup after that now. It no longer needs an ignore list from nix.
